sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -45,6 +46,25 @@ case class CreateHiveTableAsSelectCommand(
   override def innerChildren: Seq[LogicalPlan] = Seq(query)
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
+
+    // relation should move partition columns to the last
+    val (partOutputs, nonPartOutputs) = query.output.partition {
+      a =>
+        tableDesc.partitionColumnNames.contains(a.name)
+    }
+
+    // the CTAS's SELECT partition-outputs order should be consistent with
+    // tableDesc.partitionColumnNames
+    val reorderPartOutputs = tableDesc.partitionColumnNames.map {
```

Comment:
nit: `reorderPartOutputs` -> `reorderedPartOutputs`. The former sounds like a verb while the later sounds like a noun.